### PR TITLE
Add reusable build gha workflows for clojure team

### PR DESCRIPTION
**Pull request by robert-pierce**
[Card](https://rentpath.atlassian.net/browse/SRV-5149)

### CODE REVIEW

File: .github/workflows/clj-build.yml [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+          echo "sha=${SHA}" >> $GITHUB_OUTPUT
+          SHORT_SHA=$( git rev-parse --short HEAD )
+          echo "SHORT_SHA=${SHORT_SHA}" >> $GITHUB_ENV
+          echo "short_sha=${SHORT_SHA}" >> $GITHUB_OUTPUT
+          VERSION=$( date -u '+%Y%m%d' )-${{ github.run_number || 0 }}-$( git rev-parse --short HEAD )
+          echo "VERSION=${VERSION}" >> $GITHUB_ENV
+          echo "version=${VERSION}" >> $GITHUB_OUTPUT
+
+      - name: download artifacts
+        uses: actions/download-artifact@v3
```
Comment (robert-pierce) [inline .github/workflows/clj-build.yml:83]: Downloads any artifacts that are injected by the apps build action. We will need to update the Dockerfiles for these apps that have artifacts bc the directory structure has slightly changed.

File: .github/workflows/clj-build.yml [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+      - name: remove 'ready' label
+        if: github.event.pull_request && github.event.pull_request.number
+        run: |
+          curl -X DELETE \
+               -H "Accept: application/vnd.github.v3+json" \
+               -H "Authorization: bearer ${{ github.token }}" \
+               "https://api.github.com/repos/${{ github.repository}}/issues/${{ github.event.pull_request.number }}/labels/PR%3A%20RE%20Ready"
+
+      - name: check out most recent commit
+        uses: actions/checkout@v2
```
Comment (Kevinn09) [inline .github/workflows/clj-build.yml:32]: We should update this to `actions/checkout@v3`. Should work as is.

File: .github/workflows/clj-build.yml [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        run: |
+          git config --global user.email "idg@rentpath.com"
+          git config --global user.name "rentpath-plateng"
+          git tag -a ${{ steps.vars.outputs.version }} -m "${{ steps.vars.outputs.version }}" && \
+            git push origin refs/tags/${{ steps.vars.outputs.version }}:refs/tags/${{ steps.vars.outputs.version }}
+
+      - name: release
+        if: contains(github.ref, 'refs/heads/${{ inputs.target_branch }}')
+        id: create_release
+        uses: actions/create-release@v1
```
Comment (Kevinn09) [inline .github/workflows/clj-build.yml:134]: This version has been archived and contains deprecated commands. The [related repo](https://github.com/actions/create-release) provides some alternatives (I saw the [elixir team uses](https://github.com/rentpath/actions-services-workflows/blob/main/.github/workflows/deployment-images.yaml#L133) `softprops/action-gh-release@v1` - might need to update the _with_ parameters.

File: .github/workflows/clj-build.yml [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+        required: false
+        type: string
+        default: main
+jobs:
+  build:
+    name: Build App
+    runs-on: ${{ inputs.runner }}
+    steps:
+      - name: cancel prior runs
+        uses: styfle/cancel-workflow-action@0.9.1
```
Comment (Kevinn09) [inline .github/workflows/clj-build.yml:19]: This version is using Node.js 12 actions which are deprecated. I think the latest version of this action did update to Node.js 16 but it also warns towards using the native _concurrency_ property to cancel workflows (https://github.com/styfle/cancel-workflow-action) - your call.

Comment (robert-pierce) [reply .github/workflows/clj-build.yml]: Added the concurrency approach and then triggered two workflows in quick succession for sfmc-sink. The first one was canceled in favor of the latter as expected
image

File: .github/workflows/clj-build.yml [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+        required: false
+        type: string
+        default: main
+jobs:
+  build:
+    name: Build App
+    runs-on: ${{ inputs.runner }}
+    concurrency:
+      group: ${{ github.workflow }}-${{ github.ref }}
+      cancel-in-progress: ${{ github.ref != 'refs/heads/main' }}
```
[the remark this answers is not in the document]
Comment (robert-pierce) [reply .github/workflows/clj-build.yml:19]: yup ... you're correct

**Review by Kevinn09 — APPROVED** [review]

**Pull request merged**